autoload/lsp/completion.vim

```diff
@@ -92,9 +92,22 @@ export def CompletionReply(lspserver: dict<any>, cItems: any)
     lspserver.completeItemsIsIncomplete = cItems.isIncomplete
   endif
 
+  # Get the keyword prefix before the current cursor column.
+  var chcol = charcol('.')
+  var starttext = chcol == 1 ? '' : getline('.')[ : chcol - 2]
+  var [prefix, start_idx, end_idx] = starttext->matchstrpos('\k*$')
+  if opt.lspOptions.completionMatcher == 'icase'
+    prefix = prefix->tolower()
+  endif
+
+  var start_col = start_idx + 1
+
   if opt.lspOptions.ultisnipsSupport
       call UltiSnips#SnippetsInCurrentScope(1)
       for [key, info] in items(g:current_ulti_dict_info)
+          if matchfuzzy([key], prefix)->empty()
```

Comment:
If you extract all the keys at once, then you can get `matchfuzzy` to sort the results based on how well they match, I'm not sure if that is needed though?

    for key in matchfuzzy(g:current_ulti_dict_info->keys(), prefix)
      var item = g:current_ulti_dict_info[key]

Reply:
thank you for recommendation add it in https://github.com/yegappan/lsp/pull/217/commits/addf39f49791ac446b4a06aabf83656c5f34b1ad